[2024-12-23T09:15:41.051+0000] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2024-12-23T09:15:41.094+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: streaming_pipeline.image_to_datalake manual__2024-12-23T09:15:37.494503+00:00 [queued]>
[2024-12-23T09:15:41.107+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: streaming_pipeline.image_to_datalake manual__2024-12-23T09:15:37.494503+00:00 [queued]>
[2024-12-23T09:15:41.107+0000] {taskinstance.py:2306} INFO - Starting attempt 1 of 1
[2024-12-23T09:15:41.124+0000] {taskinstance.py:2330} INFO - Executing <Task(PythonOperator): image_to_datalake> on 2024-12-23 09:15:37.494503+00:00
[2024-12-23T09:15:41.132+0000] {standard_task_runner.py:63} INFO - Started process 2557 to run task
[2024-12-23T09:15:41.135+0000] {standard_task_runner.py:90} INFO - Running: ['***', 'tasks', 'run', 'streaming_pipeline', 'image_to_datalake', 'manual__2024-12-23T09:15:37.494503+00:00', '--job-id', '77', '--raw', '--subdir', 'DAGS_FOLDER/pipline_stream.py', '--cfg-path', '/tmp/tmpya3apujg']
[2024-12-23T09:15:41.137+0000] {standard_task_runner.py:91} INFO - Job 77: Subtask image_to_datalake
[2024-12-23T09:15:41.202+0000] {task_command.py:426} INFO - Running <TaskInstance: streaming_pipeline.image_to_datalake manual__2024-12-23T09:15:37.494503+00:00 [running]> on host 0b4a5bf62610
[2024-12-23T09:15:41.332+0000] {taskinstance.py:2648} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='streaming_pipeline' AIRFLOW_CTX_TASK_ID='image_to_datalake' AIRFLOW_CTX_EXECUTION_DATE='2024-12-23T09:15:37.494503+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2024-12-23T09:15:37.494503+00:00'
[2024-12-23T09:15:41.334+0000] {taskinstance.py:430} INFO - ::endgroup::
[2024-12-23T09:15:41.361+0000] {logging_mixin.py:188} INFO - Bucket raw already exists, skip creating!
[2024-12-23T09:15:41.364+0000] {conn.py:362} INFO - <BrokerConnection node_id=bootstrap-0 host=broker:29092 <connecting> [IPv4 ('172.19.0.3', 29092)]>: connecting to broker:29092 [('172.19.0.3', 29092) IPv4]
[2024-12-23T09:15:41.365+0000] {conn.py:957} INFO - Probing node bootstrap-0 broker version
[2024-12-23T09:15:41.367+0000] {conn.py:393} INFO - <BrokerConnection node_id=bootstrap-0 host=broker:29092 <connecting> [IPv4 ('172.19.0.3', 29092)]>: Connection complete.
[2024-12-23T09:15:41.473+0000] {conn.py:1019} INFO - Broker version identified as 2.6.0
[2024-12-23T09:15:41.474+0000] {conn.py:1020} INFO - Set configuration api_version=(2, 6, 0) to skip auto check_version requests on startup
[2024-12-23T09:15:41.476+0000] {consumer.py:122} WARNING - group_id is None: disabling auto-commit.
[2024-12-23T09:15:41.477+0000] {subscription_state.py:167} INFO - Updating subscribed topics to: ('raw_image',)
[2024-12-23T09:15:41.478+0000] {logging_mixin.py:188} INFO - Start consuming messages...
[2024-12-23T09:15:41.481+0000] {subscription_state.py:253} INFO - Updated partition assignment: [('raw_image', 0)]
[2024-12-23T09:15:41.484+0000] {conn.py:362} INFO - <BrokerConnection node_id=1 host=broker:29092 <connecting> [IPv4 ('172.19.0.3', 29092)]>: connecting to broker:29092 [('172.19.0.3', 29092) IPv4]
[2024-12-23T09:15:41.485+0000] {conn.py:393} INFO - <BrokerConnection node_id=1 host=broker:29092 <connecting> [IPv4 ('172.19.0.3', 29092)]>: Connection complete.
[2024-12-23T09:15:41.486+0000] {conn.py:673} INFO - <BrokerConnection node_id=bootstrap-0 host=broker:29092 <connected> [IPv4 ('172.19.0.3', 29092)]>: Closing connection. 
[2024-12-23T09:15:42.155+0000] {logging_mixin.py:188} INFO - File đã được lưu thành công vào MinIO: image/2024/2/20240200000001.jpg
[2024-12-23T09:15:44.529+0000] {logging_mixin.py:188} INFO - File đã được lưu thành công vào MinIO: image/2024/2/20240200000002.jpg
[2024-12-23T09:15:46.789+0000] {logging_mixin.py:188} INFO - File đã được lưu thành công vào MinIO: image/2024/2/20240200000003.jpg
[2024-12-23T09:15:49.045+0000] {logging_mixin.py:188} INFO - File đã được lưu thành công vào MinIO: image/2024/2/20240200000004.jpg
[2024-12-23T09:15:51.303+0000] {logging_mixin.py:188} INFO - File đã được lưu thành công vào MinIO: image/2024/2/20240200000005.jpg
[2024-12-23T09:15:53.562+0000] {logging_mixin.py:188} INFO - File đã được lưu thành công vào MinIO: image/2024/2/20240200000006.jpg
[2024-12-23T09:15:55.818+0000] {logging_mixin.py:188} INFO - File đã được lưu thành công vào MinIO: image/2024/2/20240200000007.jpg
[2024-12-23T09:15:58.085+0000] {logging_mixin.py:188} INFO - File đã được lưu thành công vào MinIO: image/2024/2/20240200000008.jpg
[2024-12-23T09:16:00.347+0000] {logging_mixin.py:188} INFO - File đã được lưu thành công vào MinIO: image/2024/2/20240200000009.jpg
[2024-12-23T09:16:02.641+0000] {logging_mixin.py:188} INFO - File đã được lưu thành công vào MinIO: image/2024/2/20240200000010.jpg
[2024-12-23T09:16:04.904+0000] {logging_mixin.py:188} INFO - File đã được lưu thành công vào MinIO: image/2024/2/20240200000011.jpg
[2024-12-23T09:16:07.159+0000] {logging_mixin.py:188} INFO - File đã được lưu thành công vào MinIO: image/2024/2/20240200000012.jpg
[2024-12-23T09:16:09.438+0000] {logging_mixin.py:188} INFO - File đã được lưu thành công vào MinIO: image/2024/2/20240200000013.jpg
[2024-12-23T09:16:11.749+0000] {logging_mixin.py:188} INFO - File đã được lưu thành công vào MinIO: image/2024/2/20240200000014.jpg
[2024-12-23T09:16:14.008+0000] {logging_mixin.py:188} INFO - File đã được lưu thành công vào MinIO: image/2024/2/20240200000015.jpg
[2024-12-23T09:16:16.266+0000] {logging_mixin.py:188} INFO - File đã được lưu thành công vào MinIO: image/2024/2/20240200000016.jpg
[2024-12-23T09:16:18.537+0000] {logging_mixin.py:188} INFO - File đã được lưu thành công vào MinIO: image/2024/2/20240200000017.jpg
[2024-12-23T09:16:20.804+0000] {logging_mixin.py:188} INFO - File đã được lưu thành công vào MinIO: image/2024/2/20240200000018.jpg
[2024-12-23T09:16:23.074+0000] {logging_mixin.py:188} INFO - File đã được lưu thành công vào MinIO: image/2024/2/20240200000019.jpg
[2024-12-23T09:16:25.377+0000] {logging_mixin.py:188} INFO - File đã được lưu thành công vào MinIO: image/2024/2/20240200000020.jpg
[2024-12-23T09:16:27.649+0000] {logging_mixin.py:188} INFO - File đã được lưu thành công vào MinIO: image/2024/2/20240200000021.jpg
[2024-12-23T09:16:29.951+0000] {logging_mixin.py:188} INFO - File đã được lưu thành công vào MinIO: image/2024/2/20240200000022.jpg
[2024-12-23T09:16:32.212+0000] {logging_mixin.py:188} INFO - File đã được lưu thành công vào MinIO: image/2024/2/20240200000023.jpg
[2024-12-23T09:16:34.515+0000] {logging_mixin.py:188} INFO - File đã được lưu thành công vào MinIO: image/2024/2/20240200000024.jpg
[2024-12-23T09:16:36.813+0000] {logging_mixin.py:188} INFO - File đã được lưu thành công vào MinIO: image/2024/2/20240200000025.jpg
[2024-12-23T09:16:39.077+0000] {logging_mixin.py:188} INFO - File đã được lưu thành công vào MinIO: image/2024/2/20240200000026.jpg
[2024-12-23T09:16:41.335+0000] {logging_mixin.py:188} INFO - File đã được lưu thành công vào MinIO: image/2024/2/20240200000027.jpg
[2024-12-23T09:16:43.602+0000] {logging_mixin.py:188} INFO - File đã được lưu thành công vào MinIO: image/2024/2/20240200000028.jpg
[2024-12-23T09:16:45.851+0000] {logging_mixin.py:188} INFO - File đã được lưu thành công vào MinIO: image/2024/2/20240200000029.jpg
[2024-12-23T09:16:48.106+0000] {logging_mixin.py:188} INFO - File đã được lưu thành công vào MinIO: image/2024/2/20240200000030.jpg
[2024-12-23T09:16:50.376+0000] {logging_mixin.py:188} INFO - File đã được lưu thành công vào MinIO: image/2024/2/20240200000031.jpg
[2024-12-23T09:16:52.639+0000] {logging_mixin.py:188} INFO - File đã được lưu thành công vào MinIO: image/2024/2/20240200000032.jpg
[2024-12-23T09:16:54.904+0000] {logging_mixin.py:188} INFO - File đã được lưu thành công vào MinIO: image/2024/2/20240200000033.jpg
[2024-12-23T09:16:57.177+0000] {logging_mixin.py:188} INFO - File đã được lưu thành công vào MinIO: image/2024/2/20240200000034.jpg
[2024-12-23T09:16:59.439+0000] {logging_mixin.py:188} INFO - File đã được lưu thành công vào MinIO: image/2024/2/20240200000035.jpg
[2024-12-23T09:17:01.714+0000] {logging_mixin.py:188} INFO - File đã được lưu thành công vào MinIO: image/2024/2/20240200000036.jpg
[2024-12-23T09:17:04.007+0000] {logging_mixin.py:188} INFO - File đã được lưu thành công vào MinIO: image/2024/2/20240200000037.jpg
[2024-12-23T09:17:06.269+0000] {logging_mixin.py:188} INFO - File đã được lưu thành công vào MinIO: image/2024/2/20240200000038.jpg
[2024-12-23T09:17:07.070+0000] {local_task_job_runner.py:310} WARNING - State of this instance has been externally set to None. Terminating instance.
[2024-12-23T09:17:07.072+0000] {local_task_job_runner.py:222} INFO - ::endgroup::
[2024-12-23T09:17:07.074+0000] {process_utils.py:132} INFO - Sending 15 to group 2557. PIDs of all processes in the group: [2557]
[2024-12-23T09:17:07.075+0000] {process_utils.py:87} INFO - Sending the signal 15 to group 2557
[2024-12-23T09:17:07.075+0000] {taskinstance.py:2611} ERROR - Received SIGTERM. Terminating subprocesses.
[2024-12-23T09:17:07.076+0000] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2024-12-23T09:17:07.087+0000] {taskinstance.py:2905} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 465, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 432, in _execute_callable
    return execute_callable(context=context, **execute_callable_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 401, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 235, in execute
    return_value = self.execute_callable()
                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 252, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/airflow/dags/stream_processing/streaming_image_to_dl.py", line 59, in image_to_dl
    for message in consumer:
  File "/home/airflow/.local/lib/python3.12/site-packages/kafka/consumer/group.py", line 1203, in __next__
    return self.next_v2()
           ^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/kafka/consumer/group.py", line 1211, in next_v2
    return next(self._iterator)
           ^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/kafka/consumer/group.py", line 1126, in _message_generator_v2
    record_map = self.poll(timeout_ms=timeout_ms, update_offsets=False)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/kafka/consumer/group.py", line 663, in poll
    records = self._poll_once(remaining, max_records, update_offsets=update_offsets)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/kafka/consumer/group.py", line 712, in _poll_once
    self._client.poll(timeout_ms=timeout_ms)
  File "/home/airflow/.local/lib/python3.12/site-packages/kafka/client_async.py", line 601, in poll
    self._poll(timeout / 1000)
  File "/home/airflow/.local/lib/python3.12/site-packages/kafka/client_async.py", line 633, in _poll
    ready = self._selector.select(timeout)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/selectors.py", line 468, in select
    fd_event_list = self._selector.poll(timeout, max_ev)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 2613, in signal_handler
    raise AirflowTaskTerminated("Task received SIGTERM signal")
airflow.exceptions.AirflowTaskTerminated: Task received SIGTERM signal
[2024-12-23T09:17:07.098+0000] {taskinstance.py:1206} INFO - Marking task as FAILED. dag_id=streaming_pipeline, task_id=image_to_datalake, run_id=manual__2024-12-23T09:15:37.494503+00:00, execution_date=20241223T091537, start_date=20241223T091541, end_date=20241223T091707
[2024-12-23T09:17:07.117+0000] {standard_task_runner.py:110} ERROR - Failed to execute job 77 for task image_to_datalake ((psycopg2.errors.ForeignKeyViolation) insert or update on table "task_fail" violates foreign key constraint "task_fail_ti_fkey"
DETAIL:  Key (dag_id, task_id, run_id, map_index)=(streaming_pipeline, image_to_datalake, manual__2024-12-23T09:15:37.494503+00:00, -1) is not present in table "task_instance".

[SQL: INSERT INTO task_fail (task_id, dag_id, run_id, map_index, start_date, end_date, duration) VALUES (%(task_id)s, %(dag_id)s, %(run_id)s, %(map_index)s, %(start_date)s, %(end_date)s, %(duration)s) RETURNING task_fail.id]
[parameters: {'task_id': 'image_to_datalake', 'dag_id': 'streaming_pipeline', 'run_id': 'manual__2024-12-23T09:15:37.494503+00:00', 'map_index': -1, 'start_date': datetime.datetime(2024, 12, 23, 9, 15, 41, 94994, tzinfo=Timezone('UTC')), 'end_date': datetime.datetime(2024, 12, 23, 9, 17, 7, 98296, tzinfo=Timezone('UTC')), 'duration': 86}]
(Background on this error at: https://sqlalche.me/e/14/gkpj); 2557)
[2024-12-23T09:17:07.168+0000] {process_utils.py:80} INFO - Process psutil.Process(pid=2557, status='terminated', exitcode=1, started='09:15:40') (2557) terminated with exit code 1
